src/backend/distributed/planner/distributed_planner.c

```diff
@@ -1536,6 +1537,24 @@ PopPlannerRestrictionContext(void)
 }
 
 
+/*
+ * ResetPlannerRestrictionContext resets the element of the given planner
+ * restriction context.
+ */
+static void
+ResetPlannerRestrictionContext(PlannerRestrictionContext *plannerRestrictionContext)
```

Comment:
It might make sense to also call this in `CreateAndPushPlannerRestrictionContext`